mlflow/metrics/genai/model_utils.py

```diff
@@ -99,66 +98,45 @@ def _call_openai_api(openai_uri, payload, eval_parameters, extra_headers, proxy_
     from mlflow.utils.openai_utils import _OAITokenHolder
 
     api_config = _get_api_config()
     api_token = _OAITokenHolder(api_config.api_type)
-
-    payload = {
-        "messages": [{"role": "user", "content": payload}],
-        **eval_parameters,
-    }
-
-    if api_config.api_type in ("azure", "azure_ad", "azuread"):
-        api_base = getattr(api_config, "api_base")
-        api_version = getattr(api_config, "api_version")
-        engine = getattr(api_config, "engine")
-        deployment_id = getattr(api_config, "deployment_id")
-
-        if engine:
-            # Avoid using both parameters as they serve the same purpose
-            # Invalid inputs:
-            #   - Wrong engine + correct/wrong deployment_id
-            #   - No engine + wrong deployment_id
-            # Valid inputs:
-            #   - Correct engine + correct/wrong deployment_id
-            #   - No engine + correct deployment_id
-            if deployment_id is not None:
-                _logger.warning(
-                    "Both engine and deployment_id are set. Using engine as it takes precedence."
-                )
-            payload = {"engine": engine, **payload}
-        elif deployment_id is None:
-            raise MlflowException(
-                "Either engine or deployment_id must be set for Azure OpenAI API",
-            )
-        payload = payload
-
-        request_url = proxy_url or (
-            f"{api_base}/openai/deployments/{deployment_id}"
-            f"/chat/completions?api-version={api_version}"
+    api_token.refresh()
+
+    if api_config.api_type == "azure":
+        from openai import AzureOpenAI
+
+        client = AzureOpenAI(
+            api_key=api_token.token,
+            azure_endpoint=proxy_url or api_config.api_base,
+            api_version=api_config.api_version,
+            azure_deployment=api_config.deployment_id,
+            max_retries=api_config.max_retries,
+            timeout=api_config.timeout,
         )
     else:
-        payload = {"model": openai_uri, **payload}
-        request_url = proxy_url or REQUEST_URL_CHAT
+        from openai import OpenAI
+
+        client = OpenAI(
+            api_key=api_token.token,
+            base_url=proxy_url or api_config.api_base,
+            max_retries=api_config.max_retries,
+            timeout=api_config.timeout,
+        )
 
     try:
-        resp = process_api_requests(
-            [payload],
-            request_url,
-            api_token=api_token,
-            throw_original_error=True,
-            max_workers=1,
+        response = client.chat.completions.create(
+            messages=[{"role": "user", "content": payload}],
+            model=openai_uri,
             extra_headers=extra_headers,
-        )[0]
-    except MlflowException as e:
-        raise e
+            **eval_parameters,
+        )
+        # to_dict is not available before openai v1.17.0
+        return _parse_chat_response_format(response.model_dump())
```

Comment:
Do we need to call `model_dump`? Can we return `resp.choices[0].message.content`?

Reply:
We could, just wanted to reuse _parse_chat_response_format if possible and this object response is harder to mimic in tests as well.

Reply:
I see, can you add a TODO comment here? I think we can use the mock openai server to simplify tests.

Reply:
Can we add a TODO comment? I think we can use the mock openai server to remove mocking.